### DISCUSSION TITLE
IFC Open Shell Python: Finding Center (x,y,z) of IfcBeam/Column

### DISCUSSION THREAD

**Discussion by RMurphySST** [IfcOpenShell/IfcOpenShell#2648] [Q&A] [answered]
Hi,
So I've been trying to write a script with IFCOpenShell in Python to find the center point locations of Beams and Columns in an IFC file. So far I have been able to find the center point of the face of the IfcBeam (where the Beam is originally extruded from) using the following code:

`placement_matrix = get_local_placement(product.ObjectPlacement)

 global_x = round(placement_matrix[0][3] * MM_TO_FEET_CONVERSION, 3)

 global_y = round(placement_matrix[1][3] * MM_TO_FEET_CONVERSION, 3)

 global_z = round(placement_matrix[2][3] * MM_TO_FEET_CONVERSION, 3)
`
This gives the location of where the Beam starts. Is there any other method to find where the middle point of the beam is?

**Comment by Moult** [comment, marked as answer]
You have so far discovered: https://blenderbim.org/docs-python/ifcopenshell-python/code_examples.html#get-the-xyz-coordinates-of-a-element and note that https://blenderbim.org/docs-python/ifcopenshell-python/code_examples.html#convert-to-and-from-si-units-and-project-units will be of help for your conversion factor.

To get the "middle point" you may process the entire geometry here, then use min/max on the vertices to get a bounding box then average it to get a centroid: https://blenderbim.org/docs-python/ifcopenshell-python/geometry_processing.html

This question has been asked many times so I suspect https://github.com/IfcOpenShell/IfcOpenShell/issues/1040 might need a bit of work.
